assembly/src/todo/mod.rs

```diff
@@ -0,0 +1,432 @@
+use core::borrow::Borrow;
+
+use vm_core::{code_blocks::CodeBlock, CodeBlockTable, Felt, FieldElement, Operation, Program};
+
+use crate::{
+    parsers::{self, Instruction, Node},
+    procedures::Procedure,
+    ModuleProvider, ProcedureAst, ProcedureId, ProgramAst,
+};
+
+pub struct Assembler {
+    module_provider: Box<dyn ModuleProvider>,
+    cb_table: CodeBlockTable,
+    locals: Vec<Procedure>,
+    blocks: Vec<Vec<CodeBlock>>,
+    spans: Vec<Vec<Operation>>,
+}
+
+impl Assembler {
+    pub fn new() -> Self {
+        Self {
+            module_provider: Box::new(()),
+            cb_table: CodeBlockTable::default(),
+            locals: Vec::new(),
+            blocks: Vec::new(),
+            spans: Vec::new(),
+        }
+    }
+
+    pub fn with_cb_table(mut self, cb_table: CodeBlockTable) -> Self {
+        self.cb_table = cb_table;
+        self
+    }
```

Comment:
We should get rid of this constructor as `cb_table` is not something that should be exposed publicly.

In general, we should have the same constructors here as we do in the current implementation of the `Assembler` (i.e., we should not change the public interface).

Reply:
Constructor removed